pkg/server/server.go

```diff
@@ -335,6 +362,7 @@ func (s *Server) Run(ctx context.Context) error {
 	}
 
 	apiExtensionsConfig.ExtraConfig.NewInformerFactoryFunc = func(client apiextensionsclient.Interface, resyncPeriod time.Duration) apiextensionsexternalversions.SharedInformerFactory {
+		// TODO could we use s.apiextensionsSharedInformerFactory (ignoring client & resyncPeriod) instead of creating a 2nd factory here?
```

Comment:
(No action required) Seems like a win to not have duplicative informer caches in the same process. What would be the drawback?

Reply:
The `NewInformerFactoryFunc` was something I added in our Kube fork so we could override the shared informer factory the apiextensions apiserver uses (so we can do workspace API inheritance). At the time, I had to do it this way because there was no way for me to inject the shared informer factory (or the client) before the server chain was started. But now that #385 is in, we can probably switch to passing in a client & shared informer factory and get rid of `NewInformerFactoryFunc`.

cc @sttts WDYT?

Reply:
Hmm looks like I'll need to do a small PR to our Kube fork to be able to pass in a client, but I could do that

Reply:
Good candidate for a follow-up PR (as it'll require a Kube PR+merge, then bumping Kube in kcp go.mod)